syntax = "proto3";

package kvstore;

service KVStoreService {
  rpc Put(PutRequest) returns (PutResponse);
  rpc Get(GetRequest) returns (GetResponse);
  rpc Heartbeat(HeartbeatRequest) returns (HeartbeatResponse);
}

message PutRequest {
  string key = 1;
  string value = 2;
}

message PutResponse {
  bool success = 1;
}

message GetRequest {
  string key = 1;
}

message GetResponse {
  string value = 1;
  bool found = 2;
}

message HeartbeatRequest {
  string node_id = 1;
}

message HeartbeatResponse {
  bool acknowledged = 1;
}